ppx/ppx_attribute_value.ml

```diff
@@ -335,30 +335,20 @@ let offset =
   let bad_form name loc =
     Ppx_common.error loc "Value of %s must be a number or percentage" name in
 
-  let regexp = Re_str.regexp "\\([-+0-9eE.]+\\)$\\|\\([0-9]+\\)%" in
+  let regexp = Re_str.regexp "\\([-+0-9eE.]+\\)\\(%\\)?" in
```

Comment:
I have a branch from weeks ago where I switched to Re combinators anyway. Can rebase, update, and PR, if you want. It will be much easier to be sure that it's correct, now that there is a test suite.